pkg/manifests/config.go

```diff
@@ -189,13 +189,14 @@ func (cps CollectionProfiles) String() string {
 	return sb.String()
 }
 
-func NewConfig(content io.Reader, collectionProfilesFeatureGateEnabled bool) (*Config, error) {
+func NewConfig(content []byte, collectionProfilesFeatureGateEnabled bool) (*Config, error) {
```

Comment:
(suggestion) let's hide the functions/methods which aren't used anymore by other packages.
```suggestion
func newConfig(content []byte, collectionProfilesFeatureGateEnabled bool) (*Config, error) {
```